### PR TITLE
Fixed GAUGE32 integer overflow in base_client.rb

### PR DESCRIPTION
**Pull request by pprugger**
As discussed here:
https://github.com/logstash-plugins/logstash-input-snmp/issues/64

Best regards

### CODE REVIEW

**Comment by pprugger** [comment]
I changed the dist to trusty, only way i found to get it building with oraclejdk8 again.

**Comment by colinsurprenant** [comment]
Thanks a lot @pprugger !  
- Could you remove the .travis.yml change, this is really non related to this PR; I will push a new PR shortly which will solve this, you can then rebase to pick it up and tests should pass.
- Would you mind adding a test/spec for this? you could add it into `spec/inputs/snmp/base_client_spec.rb` in the coercion section. Just add a test with a value that overflows without the patch and works with it. Let me know if you need help for that!

**Comment by colinsurprenant** [comment]
FYI build fixed in #66

**Comment by pprugger** [comment]
I hope it's not too much of chaos now, had some problems with rebaseing.
To be honest i have no clue how to add the test.
I would test with uint_max so 4294967295, which should be 4294967295 if it works, otherwise it should be -1. (if the overflow occurs)

**Comment by colinsurprenant** [comment]
@pprugger thanks for the dist revert. I'll send an example spec which will fail locally for me without patch and works for you, you can then include it in your PR!

**Comment by colinsurprenant** [comment]
@pprugger here, you can add these 2 tests in `spec/inputs/snmp/base_client_spec.rb" in the "coercion" context:
```
      it "should handle max unsigned 32 bits integer GAUGE32" do
        MAX_UNSIGNED_INT_32 = 4294967295
        v = Gauge32.new(MAX_UNSIGNED_INT_32)
        expect(subject.coerce(v)).to eq(MAX_UNSIGNED_INT_32)
      end

      it "should handle max signed 32 bits integer INTEGER32" do
        MAX_SIGNED_INT_32 = 2147483647
        v = Integer32.new(MAX_SIGNED_INT_32)
        expect(subject.coerce(v)).to eq(MAX_SIGNED_INT_32)
      end
```
The first one for GAUGE32 does not pass for me and returns -1 as you noted and should pass or you, the second for INTEGER32 does pass which is normal; I just added it for the sake of completeness.

**Comment by pprugger** [comment]
Did you change something else? It seems the names Gauge32 and Integer32 are not known in my context.

**Comment by pprugger** [comment]
I added the java imports. Works now and can be merged if you want.

File: spec/inputs/snmp/base_client_spec.rb [hunk 1]
```diff
@@ -32,6 +34,18 @@ def coerce(*args)
         v = AbstractVariable.create_from_syntax(SMIConstants::EXCEPTION_END_OF_MIB_VIEW )
         expect(subject).to receive(:logger).exactly(1).times.and_call_original
         expect(subject.coerce(v)).to eq("error: unknown variable syntax 130, EndOfMibView")
+      end
+	  
+	  it "should handle max unsigned 32 bits integer GAUGE32" do
```
Comment (colinsurprenant) [inline spec/inputs/snmp/base_client_spec.rb]: cosmetic: the identation is off

**Comment by colinsurprenant** [comment]
@pprugger great stuff!! yeah, forgot to mention the java imports but you figured it out 👍 
All good, minus a small cosmetic nitpick!

File: spec/inputs/snmp/base_client_spec.rb (hunk 1 above)
Comment (colinsurprenant) [inline spec/inputs/snmp/base_client_spec.rb]:
```suggestion
      it "should handle max unsigned 32 bits integer GAUGE32" do
```

**Comment by pprugger** [comment]
Fixed!

Will you do a new release after the merge?

**Comment by colinsurprenant** [comment]
Thanks a lot again for your contribution @pprugger. Feel free to submit a PR to add your information into the `CONTRIBUTORS` file.
I will merge this, bump the version and then publish it shortly.

**Comment by colinsurprenant** [comment]
v1.2.1 published.

**Pull request merged**